hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java

```diff
@@ -1386,13 +1390,12 @@ private void logPromotionCommitFailure(Address destination, Collection<Migration
                             + " since this node is shutting down.");
                     return;
                 }
-                if (logger.isFinestEnabled()) {
-                    logger.warning("Promotion commit failed for " + migrations
-                            + " since destination " + destination + " left the cluster");
-                } else {
-                    logger.warning("Promotion commit failed for "
-                            + (migrationsSize == 1 ? migrations.iterator().next() : migrationsSize + " migrations")
-                            + " since destination " + destination + " left the cluster");
+                Level level = logger.isFinestEnabled() ? FINEST : WARNING;
```

Comment:
can you please explain logic behind this?
I guess the original intent was: When FINEST logging is enabled then log more details. but it does not seems to be the case. maybe @mdogan or @metanet know better?

Reply:
ouch, I guess I need more coffee...

Reply:
Yes, we are logging all `migrations` list when finest is enabled. Otherwise just size of the migrations.